### DISCUSSION TITLE
How to config RUSTUP_TOOLCHAIN environment variable using config?

### DISCUSSION THREAD

**Discussion by yanshay** [mrcjkb/rustaceanvim#476] [Q&A]
In https://rust-analyzer.github.io/manual.html#toolchain it's explained that to override rust-analyzer toolchain, in vscode need to configure using:
`{ "rust-analyzer.server.extraEnv": { "RUSTUP_TOOLCHAIN": "stable" } }".`

I'm using `LazyVim` with `rustaceanvim` so tried the following, but it doesn't have any effect:

```
return {
  "mrcjkb/rustaceanvim",
  ft = { "rust" },
  opts = {
    server = {
      default_settings = {
        -- rust-analyzer language server configuration
        ["rust-analyzer"] = {
          server = {
            extraEnv = { ["RUSTUP_TOOLCHAIN"] = "stable" // <--------- added this
            },
          }
        },
      },
    },
  },
}
 ```

If I set the environment variable globally in the system it does work, but not through this configuration.

How can I make it work using a configuration like the above?

**Comment by mrcjkb** [comment]
Hey 👋 

rustaceanvim doesn't have a `setup` function, so you can't use lazy.nvim's `opts` table to configure it. 

In LazyVim, you should be able to use `opts` to configure rustaceanvim if you have the [rust extras](https://www.lazyvim.org/extras/lang/rust) enabled.

**Reply by yanshay** [reply]
I have the rust extras enabled, and then in order to configure the opts in lazyvim I add a file that includes the above config with the opts inside. Isn't that the right way to do it? (I'm doing it with many other plugins)

**Reply by mrcjkb** [reply]
I don't use LazyVim and don't have the capacity to provide support for Neovim distributions.
You're probably better off asking in the LazyVim discussions.
